src/provisioner/provisioner.py

```diff
@@ -256,7 +256,7 @@ def _provision_components(cortx_conf_url: str, _conf_idx: str, interfaces: Enum,
     @staticmethod
     def _apply_consul_config(_conf_idx: str):
         num_endpoints = Conf.get(_conf_idx, 'cortx>external>consul>num_endpoints')
-        for idx in range(0, num_endpoints):
+        for idx in range(0, int(num_endpoints)):
```

Comment:
instead of int at the use num_endpoints should be int. So better code would be
``` 
try:
  num_endpoints = int(Conf.get(_conf_idx, 'cortx>external>consul>num_endpoints'))
  for idx in range(0, num_endpoints):
    ...
except ConfError:
  raise ...
```

Reply:
comments addressed